core/storage/user/gae_models_test.py

```diff
@@ -84,6 +84,127 @@ def test_get_failure(self):
+        user_model_1.impact_score = self.USER_1_IMPACT_SCORE
+        user_model_1.total_plays = self.USER_1_TOTAL_PLAYS
+        user_model_1.average_ratings = self.USER_1_AVERAGE_RATINGS
+        user_model_1.num_ratings = self.USER_1_NUM_RATINGS
+        user_model_1.weekly_creator_stats_list = (
+            self.USER_1_WEEKLY_CREATOR_STATS_LIST)
+        user_models.UserStatsModel.put(user_model_1)
+
+        user_model_2 = user_models.UserStatsModel(id=self.USER_ID_2)
+        user_model_2.impact_score = self.USER_2_IMPACT_SCORE
+        user_model_2.total_plays = self.USER_2_TOTAL_PLAYS
+        user_model_2.average_ratings = self.USER_2_AVERAGE_RATINGS
+        user_model_2.num_ratings = self.USER_2_NUM_RATINGS
+        user_model_2.weekly_creator_stats_list = (
+            self.USER_2_WEEKLY_CREATOR_STATS_LIST)
+        user_models.UserStatsModel.put(user_model_2)
+
+    def test_export_data_on_existing_user(self):
+        """Test if export_data works when user is in data store."""
+        user_data = user_models.UserStatsModel.export_data(self.USER_ID_1)
+        test_data = {
+            'impact_score': self.USER_1_IMPACT_SCORE,
+            'total_plays': self.USER_1_TOTAL_PLAYS,
+            'average_ratings': self.USER_1_AVERAGE_RATINGS,
+            'num_ratings': self.USER_1_NUM_RATINGS,
+            'weekly_creator_stats_list': self.USER_1_WEEKLY_CREATOR_STATS_LIST
+        }
+        self.assertEqual(user_data, test_data)
+
+    def test_export_data_on_multiple_users(self):
+        """Test if export_data works on multiple users in data store."""
+        user_1_data = user_models.UserStatsModel.export_data(self.USER_ID_1)
+        test_1_data = {
+            'impact_score': self.USER_1_IMPACT_SCORE,
+            'total_plays': self.USER_1_TOTAL_PLAYS,
+            'average_ratings': self.USER_1_AVERAGE_RATINGS,
+            'num_ratings': self.USER_1_NUM_RATINGS,
+            'weekly_creator_stats_list': self.USER_1_WEEKLY_CREATOR_STATS_LIST
+        }
+
+        user_2_data = user_models.UserStatsModel.export_data(self.USER_ID_2)
+        test_2_data = {
+            'impact_score': self.USER_2_IMPACT_SCORE,
+            'total_plays': self.USER_2_TOTAL_PLAYS,
+            'average_ratings': self.USER_2_AVERAGE_RATINGS,
+            'num_ratings': self.USER_2_NUM_RATINGS,
+            'weekly_creator_stats_list': self.USER_2_WEEKLY_CREATOR_STATS_LIST
+        }
+
+        self.assertEqual(user_1_data, test_1_data)
+        self.assertEqual(user_2_data, test_2_data)
+
+    def test_export_data_on_nonexistent_user(self):
+        """Test if export_data returns None when user is not in data store."""
+        user_data = user_models.UserStatsModel.export_data(self.USER_ID_3)
+        test_data = None
+        self.assertEqual(user_data, test_data)
+
+    def test_export_data_stats_list_is_well_formed(self):
+        """Test if weekly creator stats list has only the expected elements."""
```

Comment:
Isn't this test essentially a subset of test_export_data_on_existing_user()? Why do we need it / what information does it give us that that other test does not?

Reply:
PTAL @seanlip  at the updated version (one line changed).

**What I was getting at**: export_data explicitly constructs the stats list*, so the purpose of this test was to make sure that the weekly_creator_stats_list in the class (not the export_data) contains the expected elements. Otherwise, it's possible that we did not fully export all the data in the stats list since someone decided to add more entries to the stats list (entries like 'total_plays'). This test will catch that change and prompt whoever is adding more data to git-blame the file and get in contact with me or someone else so we can properly update export_data to include their new features. 

**What was wrong**: I used export_data, not the weekly_creator_stats_list of the user model from the data store. I'll fix that--my bad.

Is this not necessary? Why?

*latest PR should do it correctly

Reply:
Is this test fine given the recent developments about how we are approaching takeout (timeline wise), or should it be dropped/modified? @seanlip

Reply:
I think it can be dropped. Would you agree?

Also, thanks for checking!

Reply:
Agreed!